core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -120,15 +118,20 @@ private FileIO io(String location) {
       try {
         io = CatalogUtil.loadFileIO(impl, properties, conf);
       } catch (IllegalArgumentException e) {
-        LOG.warn("Failed to load FileIO implementation: {}, falling back to {}", impl, FALLBACK_IMPL, e);
-        try {
-          // couldn't load the normal class, fall back to HadoopFileIO
-          io = CatalogUtil.loadFileIO(FALLBACK_IMPL, properties, conf);
-        } catch (IllegalArgumentException suppressed) {
-          LOG.warn("Failed to load FileIO implementation: {} (fallback)", FALLBACK_IMPL, suppressed);
-          // both attempts failed, throw the original exception with the later exception suppressed
-          e.addSuppressed(suppressed);
+        if (impl.equals(FALLBACK_IMPL)) {
+          // no implementation to fall back to, throw the exception
           throw e;
+        } else {
+          // couldn't load the normal class, fall back to HadoopFileIO
+          LOG.warn("Failed to load FileIO implementation: {}, falling back to {}", impl, FALLBACK_IMPL, e);
```

Comment:
Nit: Maybe `... falling back to only using ${FALLBACK_IMPL}`?

Something to really clarify that this is the only implementation that will be used. Up to you if you feel it's clear enough as is (probably it is).